linkerd/stack/src/watch.rs

```diff
@@ -48,31 +47,50 @@ pub struct SpawnWatch<S> {
 // === impl NewSpawnWatch ===
 
 impl<P, N> NewSpawnWatch<P, N> {
-    pub fn new(new_update: N) -> Self {
+    pub fn new(inner: N) -> Self {
         Self {
-            new_update,
+            inner,
             _marker: std::marker::PhantomData,
         }
     }
+
+    pub fn layer() -> impl tower::layer::Layer<N, Service = Self> + Clone {
+        crate::layer::mk(Self::new)
+    }
+
+    // Creates a new `NewSpawnWatch` layer that transforms the target type to
+    // `T` for the inner stack.
+    pub fn layer_into<T>(
+    ) -> impl tower::layer::Layer<N, Service = NewSpawnWatch<P, NewWatchedFromTuple<T, N>>> + Clone
```

Comment:
this is probably the weirdest rustfmt decision i've seen in a while, although i guess i get why it did that (not wanting to line break inside the return type)... :woman_shrugging:  probably not worth trying to make the type shorter just to avoid this...